compiler/parser/chapel.ypp

```diff
@@ -664,6 +664,39 @@ renames_ls:
                                          $1->push_back(cur); }
 ;
 
+/* Separated so that use statements can rename their module to "_", but nothing
+   else can */
```

Comment:
For this kind of thing, I find it's often less complexity if the parser allows something more general (in this case, `_` in only/except lists), and then separate code (e.g. in checks, or in build) performs an additional check on the AST (in this case, `_` is not used in only/except lists). It would seem that this would really simplify this parser change. I understand though that parser changes sometimes aren't as easy as you might hope so please let me know if this has some big challenge I do not see.

Reply:
The main reason I opted for this route is that it isn't just except/only lists that shouldn't allow `_`, it's also imported modules, their lists for unqualified access, and forwarding statements.  That's four different places that would have to add checks, all so that one place can support them.  I don't expect we'll extend to other areas, but if we did, we'd probably not want to support `_` there either, and would have to actively remember to add something to avoid supporting it (which will be harder to do since it is not in the same place as the code enabling `_`).

Maybe there's a way to refactor the parser rules so I'm not duplicating as much code?  I don't know if that would reduce complexity, though